include/filters/filter_chain.hpp

```diff
@@ -30,15 +30,124 @@
 #ifndef FILTERS_FILTER_CHAIN_HPP_
 #define FILTERS_FILTER_CHAIN_HPP_
 
-#include "ros/ros.h"
 #include "filters/filter_base.hpp"
-#include <pluginlib/class_loader.h>
+#include <pluginlib/class_loader.hpp>
+#include <algorithm>
 #include <sstream>
 #include <vector>
-#include "boost/shared_ptr.hpp"
 
 namespace filters
 {
+  namespace impl
+  {
+    struct FoundFilter {
+      std::string name;
+      std::string type;
+      std::string param_prefix;
+    };
+
+    /* @brief Read params and figure out what filters to load
+     */
+    bool
+    load_chain_config(
+      const std::string & param_prefix,
+      const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr & node_logger,
+      const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr & node_params,
+      std::vector<struct FoundFilter> & found_filters)
+    {
+      // TODO(sloretz) error if someone tries to do filter0
+
+      // Make prefix be an empty string or a string ending in '.'
+      std::string norm_param_prefix = param_prefix;
+      if (!norm_param_prefix.empty()) {
+        if ('.' != norm_param_prefix.back()) {
+          norm_param_prefix += '.';
+        }
+      }
```

Comment:
Seems like common code, at least it is duplicated here with `FilterBase`.  Maybe make a utility?  Feel free to open an issue to followup.